### DISCUSSION TITLE
Platform agnostic version links

### DISCUSSION THREAD

**Discussion by aryelgois** [olivierlacan/keep-a-changelog#485] [Ideas]
Hey, just a small question:

I am using links in the versions to point to github's compare, just like in this repository.

But they get useless if the project migrates to another platform (bitbucket, gitlab, etc)

What can I do?

It would be awesome if keep-a-changelog got really big and these platforms processed the changelogs to replace the links to their own site, when viewing the rendered markdown..

**Comment by oxr463** [comment]
Any updates on this?

**Comment by oxr463** [comment]
# GitLab

```yaml
unreleased: https://gitlab.com/{{namespace}}/{{repository}}/-/compare/{{tag}}...HEAD

release_{{version}}: https://gitlab.com/{{namespace}}/{{repository}}/-/compare/{{previous_version}}...{{version}}
```

**Comment by aryelgois** [comment]
# Bitbucket

```markdown
[unreleased]: https://bitbucket.org/{{namespace}}/{{repository}}/branches/compare/HEAD..{{version}}

[{{version}}]: https://bitbucket.org/{{namespace}}/{{repository}}/branches/compare/{{version}}..{{previous_version}}
```

**Comment by olivierlacan** [comment]
@aryelgois That's a very good idea for sure. 

It would be nice to have /{{organization}}/{{repository}}/branches/compare/{{lastVersion}}..{{currentVersion}} direct to the appropriate location regardless of hosting provider for sure. But having already bitten off a chunk of platform advocacy with shields.io a decade ago I'm not sure I want to to try and convince Atlassian, Microsoft, and GitLab to play nice with each other on this.

There's a chance this project could have sufficient leverage to advocate for this change and get their attention for sure, which is why this is an interesting idea to me. The lack of portability of these compare links has always bothered me. 

But better yet, I'd love if it we could get those links to function locally and have each platform expand them accordingly. Since the local portability of the CHANGELOG.md file is what still makes it more useful in my mind than even the now-improved release pages organizations like GitHub have built.

**Comment by aryelgois** [comment]
Like some kind of magic link, for example:

```
git://diff/{{lastVersion}}..{{currentVersion}}
```

I omitted `{{organization}}` and `{{repository}}` because:

- they are not really a thing in Git itself
- the Changelog should be about the current repository anyways
    - bonus portability for renaming the repository
- a link for an external repository should be the full canonical URL, for where it is hosted
